UserService/src/main/java/com/streamster/userservice/controller/UserController.java

```diff
@@ -3,36 +3,57 @@
 import com.streamster.userservice.model.SystemRoleType;
 import com.streamster.userservice.model.User;
 import com.streamster.userservice.model.dto.RegistrationDTO;
+import com.streamster.userservice.model.view.UserView;
+import com.streamster.userservice.repository.UserRepository;
 import com.streamster.userservice.service.UserService;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.security.access.prepost.PreAuthorize;
 import org.springframework.web.bind.annotation.*;
 
+import java.security.Principal;
+import java.util.List;
+import java.util.stream.Collectors;
+
 @RestController
-@RequestMapping("/user")
+@RequestMapping("/users")
 public class UserController {
 
     final UserService userService;
+    final UserRepository userRepository;
 
-    public UserController(UserService userService) {
+    public UserController(UserService userService, UserRepository userRepository) {
         this.userService = userService;
+        this.userRepository = userRepository;
     }
 
     @GetMapping("/test")
     ResponseEntity<String> test() {
         return new ResponseEntity<>("SUCCESS",HttpStatus.OK);
     }
 
+    @GetMapping("/userDetails")
+    ResponseEntity<UserView> getUserDetails(Principal principal) {
+        User currentUser = userService.getUserByEmail(principal.getName());
+        return new ResponseEntity<>(currentUser.toUserView(), HttpStatus.OK);
+    }
+
+    @PreAuthorize("hasAuthority(T(com.streamster.userservice.model.SystemRoleType).ADMIN)")
+    @GetMapping("/all")
```

Comment:
you don't need specific path to get all items from collection. So it's just @GetMapping, without the additional path